src/js/angular/aclmanagement/model.js

```diff
@@ -154,24 +267,187 @@ export class ACRuleModel {
+    }
+
+    get operation() {
+        return this._operation;
+    }
+
+    set operation(value) {
+        this._operation = value;
+    }
+
+    toJSON() {
+        return {
+            scope: this.scope,
+            policy: this.policy,
+            role: this.role,
+            operation: this.operation,
+            plugin: this.plugin
+        }
+    }
+}
+
+/**
+ * Represents a ClearGraphACRuleModel.
+ *
+ * @extends ACRuleModel
+ */
+export class ClearGraphACRuleModel extends ACRuleModel {
+    /**
+     * Creates a new instance of the constructor.
+     *
+     * @param {string} role - The role associated with the rule.
+     * @param {string} [policy=ACL_POLICY.ALLOW] - The policy for the role. Default value is ACL_POLICY.ALLOW.
+     * @param {Object} context - The context object.
+     */
+    constructor(role, policy, context) {
+        super(ACL_SCOPE.CLEAR_GRAPH, role, policy);
+        this._context = context;
+    }
+
+    get context() {
+        return this._context;
+    }
+
+    set context(value) {
+        this._context = value;
+    }
+
+    toJSON() {
+        return {
+            scope: this.scope,
+            policy: this.policy,
+            role: this.role,
+            context: this.context
+        }
+    }
+}
+
+/**
+ * Represents a system access control rule model.
+ * @extends ACRuleModel
+ */
+export class SystemACRuleModel extends ACRuleModel {
+    constructor(role, policy, operation) {
+        super(ACL_SCOPE.SYSTEM, role, policy);
+        this._operation = operation;
+    }
+
+    get operation() {
+        return this._operation;
+    }
+
+    set operation(value) {
+        this._operation = value;
+    }
+
+    toJSON() {
+        return {
+            scope: this.scope,
+            policy: this.policy,
+            role: this.role,
+            operation: this.operation
+        }
+    }
+}
+
+class ACRuleFactory {
+    /**
+     * Creates an access control rule based on the given parameters.
+     *
+     * @param {string} scope - The scope of the rule. Can be one of ACL_SCOPE.STATEMENT, ACL_SCOPE.PLUGIN, ACL_SCOPE.CLEAR_GRAPH, or ACL_SCOPE.SYSTEM.
+     * @param {string} policy - The policy of the rule. Can be "ALLOW" or "DENY".
+     * @param {string} role - The role associated with the rule.
+     * @param {string} operation - The operation associated with the rule.
+     * @param {string} subject - The subject associated with the rule.
+     * @param {string} predicate - The predicate associated with the rule.
+     * @param {string} object - The object associated with the rule.
+     * @param {string} context - The context associated with the rule.
+     * @param {string} plugin - The plugin associated with the rule.
+     * @returns {ACRuleModel} - The created access control rule.
+     * @throws {Error} - Throws an error if an invalid scope is provided.
+     */
+    static createRule(scope, policy, role, operation, subject, predicate, object, context, plugin) {
```

Comment:
As far as I can see, all prams except scope are optional.

Reply:
fixed the documentation